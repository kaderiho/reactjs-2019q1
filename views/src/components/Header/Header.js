"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _reactRouterDom = require("react-router-dom");

var _Button = _interopRequireDefault(require("../Button/Button"));

var Header = function Header(_ref) {
  var isSearchButtonEnabled = _ref.isSearchButtonEnabled;
  return _react.default.createElement("header", null, _react.default.createElement("span", {
    className: "logoLink"
  }, "netflixroulette"), isSearchButtonEnabled && _react.default.createElement(_reactRouterDom.Link, {
    to: "/"
  }, _react.default.createElement(_Button.default, {
    variant: "contained",
    color: "fourth",
    className: "header-searchButton"
  }, "Search")));
};

var _default = Header;
exports.default = _default;