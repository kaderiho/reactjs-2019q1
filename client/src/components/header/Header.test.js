import React from 'react';
import { shallow , mount } from 'enzyme';
import Header from './Header';

/**
 * Tests
 * 
 * - To match snapshot
 */

describe('Header component', () => {
    it('Should render correctly', () => {
        const wrapper = shallow(<Header />);
        expect(wrapper).toMatchSnapshot();
    });
});