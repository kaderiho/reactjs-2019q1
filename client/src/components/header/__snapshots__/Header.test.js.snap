// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component Should render correctly 1`] = `
<header>
  <a
    className="logoLink"
    href="#"
  >
    netflixroulette
  </a>
</header>
`;
